Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -205,14 +210,52 @@ public class NIOSSLHandler : ChannelInboundHandler, ChannelOutboundHandler, Remo
             self.doUnbufferWrites(context: context)
         }
     }
-    
+
     public func close(context: ChannelHandlerContext, mode: CloseMode, promise: EventLoopPromise<Void>?) {
-        guard mode == .all else {
-            // TODO: Support also other modes ?
+        switch mode {
+        case .output:
+            self.closeOutput(context: context, promise: promise)
+        case .all:
+            self.closeAll(context: context, promise: promise)
+        default:
             promise?.fail(ChannelError.operationUnsupported)
-            return
         }
-        
+    }
+
+    private func closeOutput(context: ChannelHandlerContext, promise: EventLoopPromise<Void>?) {
+        switch state {
+        case .closing:
+            // We're in the process of TLS shutdown, which has a higher priority.
+            // Therefore we skip the output closing procedure and cascade the result
+            // of the TLS shutdown request to this new one.
+            if let promise = promise, let closePromise = self.closePromise {
+                closePromise.futureResult.cascade(to: promise)
+            } else if let promise = promise {
+                self.closePromise = promise
+            }
+        case .idle, .closed, .unwrapping, .unwrapped:
+            // For idle, closed, unwrapping, and unwrapped connections we immediately pass this on to the next
+            // channel handler.
+            context.close(mode: .output, promise: promise)
```

Comment:
Question: what do you think of this behaviour? Passing on `context.close(mode: .output)` can result in errors if the next `ChannelHandler` does not support `CloseMode.output`?

Reply:
Passing this on is fine.